### ISSUE TITLE
Regression: on expectedExceptionsMessageRegExp expected and actual messages are not printed.

### ISSUE THREAD

**Issue by risenberg** [cbeust/testng#1409]
### TestNG Version
6.11

### Expected behavior
If test code throws exception that passes `expectedExceptions` check, but violates `expectedExceptionsMessageRegExp`, then testng says that exception message was incorrect and reports the actual and expected message.
This worked in 6.8.8 and stopped working in 6.11.

### Actual behavior
testng outputs the exception as is.

### Is the issue reproductible on runner?

- [x] IntelliJ

### Test case sample
```
    @Test(expectedExceptions = RuntimeException.class,
            expectedExceptionsMessageRegExp = "expected")
    public void test() {
        throw new RuntimeException("actual");
    }
```

**Comment by juherr** [comment]
Thanks for the report but there were many versions between 6.8.8 and 6.11.

Could you try to find in which version the issue appeared?

**Comment by risenberg** [comment]
I tested the versions available in maven central - 6.8.14 is good, 6.8.15 fails internally, 6.8.17 is bad.
